### PR TITLE
Replace old model-mesh manifests to new manifests

### PR DESCRIPTION
**Pull request by Jooho**
## Description
The old model mesh manifest was independent of the upstream model mesh, so when the upstream model mesh manifest changed, it was very difficult to update the opendatahub manifest to follow its history. So the new manifest uses the upstream manifest as much as possible via an overlay, with only minor changes. 
Inside this folder, there are now two main component manifest folders: `odh-model-controller`, `odh-modelmesh-controller`
Each component utilized the manifest from the main repository as much as possible and used overlay or params.env for other components.

## How Has This Been Tested?
The way how to use the `model-mesh` manifests is the same as before so you can test this with the kfdef that we used.
~~~
kind: KfDef
apiVersion: kfdef.apps.kubeflow.org/v1
metadata:
  name: opendatahub
  namespace: opendatahub
spec:
  applications:
    - kustomizeConfig:
        repoRef:
          name: manifests
          path: odh-common
      name: odh-common
    - kustomizeConfig:
        repoRef:
          name: manifests
          path: model-mesh
      name: model-mesh
  repos:
    - name: manifests
      uri: 'https://github.com/Jooho/odh-manifests/tarball/new-mm-manifests'
~~~

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

ISSUE: https://issues.redhat.com/browse/RHODS-8375

### CODE REVIEW

**Comment by Jooho** [comment]
@anishasthana Finally this passed all tests!. Could you please review this PR?

File: model-mesh/OWNERS [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-- anishasthana
-- crobby
+  - anishasthana
+  - Jooho
+  - Xaenalt
+  - israel-hdez
+  - heyselbi
+  - VedantMahabaleshwarkar
 
 reviewers:
```
[the remark this answers is not in the document]
Comment (Jooho) [reply model-mesh/OWNERS:9]: updated

File: model-mesh/base/params.env [hunk 2]
```diff
@@ -1,8 +1,7 @@
-meshnamespace=
-monitoring-namespace=
-odh-mm-rest-proxy=quay.io/opendatahub/rest-proxy:v0.11.0-alpha
+monitoring-namespace=opendatahub
```
[the remark this answers is not in the document]
Comment (Jooho) [reply model-mesh/base/params.env:1]: For RHODS disconnected support, kfdef will overwrite the image tag to image sha. At the moment, we don't support disconnected for open data hub. We need to revisit this sometime later.

**Comment by danielezonca** [comment]
/lgtm

Well done :)

**Review by danielezonca — APPROVED** [review]

**Review by anishasthana — APPROVED** [review]
Approving as two component owners have approved/lgtm'd it.

**Comment by shalberd** [comment]
Follow-up minor issue relared to serving runtimes: https://github.com/opendatahub-io/odh-manifests/issues/809

Inconsistent usage of kustomization.images refs vs parameter (ovms).

**Comment by shalberd** [comment]
another error that came up previous to this master-level change, fixed with the changes in this PR / master:

```
svens-mbp-2:odh-manifests-1.6.1 s$ pwd
/Users/s/Downloads/odh-manifests-1.6.1

svens-mbp-2:odh-manifests-1.6.1 s$ kustomize version

{Version:kustomize/v4.5.6 GitCommit:29ca6935bde25565795e1b4e13ca211c4aa56417 BuildDate:2022-07-29T20:42:23Z GoOs:darwin GoArch:amd64}

svens-mbp-2:odh-manifests-1.6.1 s$ kustomize build  ./model-mesh/base
Error: accumulating resources: accumulation err='accumulating resources from '../dependencies/quickstart.yaml': security; file '/Users/s/Downloads/odh-manifests-1.6.1/model-mesh/dependencies/quickstart.yaml' is not in or below '/Users/s/Downloads/odh-manifests-1.6.1/model-mesh/base'': must build at directory: '/Users/s/Downloads/odh-manifests-1.6.1/model-mesh/dependencies/quickstart.yaml': file is not directory
```

File: model-mesh/README.md [hunk 3]
```diff
@@ -39,7 +50,7 @@ Following are the steps to install Model Mesh as a part of OpenDataHub install:
 
 1. Install the OpenDataHub operator
 2. Create a KfDef that includes the model-mesh component with the odh-model-controller overlay.
```
[the remark this answers is not in the document]
Comment (Jooho) [reply model-mesh/README.md:52]: Good Catcher. Right, this should be deleted.

**Pull request merged**